pay-api/src/pay_api/services/payment.py

```diff
@@ -366,21 +382,54 @@ def create_payment_report(auth_account_id: str, search_filter: Dict,
 
         results = Payment.search_all_purchase_history(auth_account_id, search_filter)
 
-        report_response = Payment.generate_payment_report(content_type, report_name, results,
-                                                          template_name='payment_transactions')
+        report_response = Payment.generate_payment_report(PaymentReportInput(content_type=content_type,
+                                                                             report_name=report_name,
+                                                                             template_name='payment_transactions',
+                                                                             results=results
+                                                                             ))
         current_app.logger.debug('>create_payment_report')
 
         return report_response
 
+    @staticmethod
+    def get_invoices_totals(invoices: List[InvoiceModel]):
```

Comment:
This was moved from the generate report function so it can be re-used to calculate previous statement values as well.